packages/flutter/test/rendering/mock_canvas.dart

```diff
@@ -74,6 +74,15 @@ typedef void _CanvasPainterFunction(Canvas canvas);
 /// Patterns are subset matches, meaning that any calls not described by the
 /// pattern are ignored. This allows, for instance, transforms to be skipped.
 abstract class PaintPattern {
+  /// Indicates that a transform is expected next.
+  ///
+  /// Calls are skipped until a call to [Canvas.transform] is found. The call's
+  /// arguments are compared to those provided here. If any fail to match, or if
+  /// no call to [Canvas.transform] is found, then the matcher fails.
+  ///
+  /// Dynamic so matchers can be more easily passed in.
```

Comment:
This last sentence isn't clear. Does that mean they could even if it wasn't dynamic? Does it mean that matchers must be passed in?

We should have a longer paragraph that talks about what exactly can be passed in.

Ideally we wouldn't use dynamic. Maybe have `transform` and `transformMatches`, where the second takes a "Matcher"?

Reply:
You don't seem to actually ever call the matcher API, so I'm not sure how this is working.

Reply:
Added more doc following the `paragraph` call example that uses the dynamic pattern too. 

The matching is done via _FunctionPaintPredicate which checks each argument for equality or match. The deep matcher via `equals` compares the Float64List's iterable's value with more matchers.

Reply:
ah i see